command/alloc_logs.go

```diff
@@ -195,77 +208,99 @@ func (l *AllocLogsCommand) Run(args []string) int {
 
 	// If -task isn't provided fallback to reading the task name
 	// from args.
-	if task != "" {
-		err = validateTaskExistsInAllocation(task, alloc)
+	if l.task != "" {
+		err = validateTaskExistsInAllocation(l.task, alloc)
 	} else {
 		if len(args) >= 2 {
-			task = args[1]
-			if task == "" {
+			l.task = args[1]
+			if l.task == "" {
 				l.Ui.Error("Task name required")
 				return 1
 			}
 		} else {
-			task, err = lookupAllocTask(alloc)
+			l.task, err = lookupAllocTask(alloc)
 		}
 	}
 	if err != nil {
 		l.Ui.Error(fmt.Sprintf("Failed to validate task: %s", err))
 		return 1
 	}
 
-	logType := "stdout"
-	if stderr {
-		logType = "stderr"
+	// In order to run the mixed log output, we can only follow the files from
+	// their current positions. There is no way to interleave previous log
+	// lines as there is no timestamp references.
+	if !l.stderr && !l.stdout && l.follow && !l.tail && l.numLines < 0 && l.numBytes < 0 {
```

Comment:
```suggestion
	if l.follow && !(l.stderr || l.stdout || l.tail || l.numLines > 0 || l.numBytes > 0) {
```

I was a little confused with this check so I think applying a DeMorgan on the negative components makes the rule a bit more clear? You need `-f` and can't have any of the others.